### ISSUE TITLE
It's not using my pre-existing Traefik config

### ISSUE THREAD

**Issue by charl3y15** [Rahn-IT/traefik-gui#1]
Is it supposed to? I pointed to the right path but all I see is 
![image](https://github.com/user-attachments/assets/edb41832-b4df-4fda-b55f-d046a087f44b)

**Comment by charl3y15** [comment]
I have this mapping /mnt/user/appdata/traefik:/app/traefik

**Comment by acul009** [comment]
The routes are saved in an sqlite database and only exported to the traefik format.

If you‘re configs are simple enough I might be able to add an import. Do you have an example you could share?

There‘s also a few more things:
- thanks for the screenshot, I just spotted a bug in my export code.
- Https routes are not yet implemented
- The host field doesn‘t require a protocol. `panel.example.com` should be enough.

I‘m a bit busy at the moment, but I‘ll fix the bug and add Https if I find some time :)

**Comment by charl3y15** [comment]
No rush! I just noticed when I added it it didn't seem to point to the right location. Is Traefik still able to read when I add entries? for example, the panel.cd1git.com

Most of my config that I would like to be imported look like

```
    dockerwebgui:
        entryPoints:
          - https
        rule: 'Host(`dockerwebgui.domain.com`)'
        service: dockerwebgui
        middlewares:
          # - "crowdsec-bouncer@file"
          - "securityHeaders"
          - "authentik@file"
```

I am not sure if the comments would make it more difficult to import

**Comment by acul009** [comment]
So I took a stab at adding https. I still need to test it a bit more, but I think it should work ok. It will require you to specify the name of your cert-provider in the general settings, but I‘m guessing that‘s not really a problem.

As for importing your existing routes - the comments arent a problem, but the middleware is.

I never designed this thing with a more complicated configuration in mind and right now I‘m unsure how I‘d go about adding middlewares. I could just add an extra field to the routes, but that would make using this thing even harder for newcomers. I‘d actually like to work on making it easier to use to be a possible replacement for the Nginx Proxy Manager.

Another question to stisfy my curiosity: From your configs it seems like you‘re combining docker and file configs, is that correct? How does your general http routing setup look like?

Edit: to clarify: I‘m not against enabling middlewares, I‘m just unsure hoe I‘d add that to the UI in a nice way. Maybe a button to open some advanced settings on a route?

**Comment by charl3y15** [comment]
For some reason it's not updating my Traefik config, I'm using Unraid. I've mounted /app/traefik to /mnt/user/appdata/traefik

Do I need to map the db path to that as well?

Didn't have issues when using the docker compose

**Comment by acul009** [comment]
No, that shouldn't be neccesary. The DB is only used to save your settings.

Can you check that path manually?
If you press Redeploy, do you get any errors?

It could be a permission issue. Do you know which user unraid runs the container as?

**Comment by charl3y15** [comment]
Right now Unraid 7, i was doing some renovation so i can't get my server up tonight, hopefully i didn't mess any drives up lol

tbh, the Unraid version doesn't matter that much

**Comment by acul009** [comment]
If you have your setup working again and want help, just update this issue.
In the meanwhile I do have an update ready which would allow for simple https routes.
I could publish it, if you want.

**Comment by acul009** [comment]
@charl3y15 Do you still need or want help?

**Comment by charl3y15** [comment]
Let me try again in few

**Comment by charl3y15** [comment]
It's still not pulling from my existing config

**Comment by acul009** [comment]
> It's still not pulling from my existing config

traefik-gui only supports adding routes through the interface at the moment. But traefik itself supports multiple dynamic configs, so you could just add your files into the normal config dir.

I might be able to add some functionality if you can explain what you would like to do. Du you just want to manually add configs, e.g. in a text field?

**Comment by charl3y15** [comment]
Ah don't worry about it then, i thought you were trying to figure out how to import existing ones
